src/actors/builtin/storage_miner/storage_miner_actor_state.go

```diff
@@ -1,25 +1,138 @@
 package storage_miner
 
 import (
+	"math/big"
+
 	addr "github.com/filecoin-project/go-address"
 	abi "github.com/filecoin-project/specs/actors/abi"
 	indices "github.com/filecoin-project/specs/actors/runtime/indices"
-	actor_util "github.com/filecoin-project/specs/actors/util"
+	autil "github.com/filecoin-project/specs/actors/util"
+	cid "github.com/ipfs/go-cid"
 	peer "github.com/libp2p/go-libp2p-core/peer"
 )
 
-func (st *StorageMinerActorState_I) _getSectorOnChainInfo(sectorNo abi.SectorNumber) (info SectorOnChainInfo, ok bool) {
-	sectorInfo, found := st.Sectors()[sectorNo]
+// Balance of a StorageMinerActor should equal exactly the sum of PreCommit deposits
+// that are not yet returned or burned.
+type StorageMinerActorState struct {
+	Sectors    SectorsAMT
+	PoStState  MinerPoStState
+	ProvingSet SectorNumberSetHAMT
+	Info       MinerInfo
+}
+
+type MinerPoStState struct {
+	// If > 0 miner is in an OK state.
```

Comment:
I see some redundancy here. Let's say initially -1, then the epoch of the last successful post. Let's not nuke it when they enter another state. This means that  whether it's in an ok state is whether SurpriseChallengeEpoch is <0 and NumConsecutiveFailures == 0